gocourse04/SmartZoo/main.go

```diff
@@ -0,0 +1,253 @@
+/*
+При розробці системи «Розумний зоопарк» техлід ще не вирішив, яку базу даних використовувати і для реалізації
+прототипа вирішили зберігати дані в памʼяті програми, а саме — в мапах. Треба побудувати складну структуру мап,
+відобразивши деяку ієрархію частин зоопарку. Наприклад, зоопарк ділиться на території: копитні, пернаті,
+примати… та інше. Кожна територія має декілька загонів для кожного виду тварин. Кожен загін може мати ділянку
+для перебування тварини і технічне приміщення, де можуть зберігатися приладдя для роботи з тваринами. На кожній
+території є певна кількість тварин, а технічне приміщення «здатне» виконувати деякі функції, такі як прибирання і
+годування тварин. Реалізувати функції пошуку тварини за імʼям або ID, переміщення тварини із загону в загін,
+годування тварини.
+*/
+package main
+
+import (
+	"errors"
+	"fmt"
+)
+
+type Areas map[string]Area
+
+// Територія по  типу (копитні, пернаті, примати)
+type Area struct {
+	Name    string
+	Sectors map[string]Sector
+}
+
+// Загон для виду тварин. У кожному загоні є технічне приміщення (наприклад годування)
+type Sector struct {
+	ID      int
+	Animals []Animal
+}
+
+type Animal struct {
+	ID   int
+	Name string
+}
+
+type Zoo struct {
+	Areas Areas
+}
+
+func NewZoo() *Zoo {
+	return &Zoo{Areas: AreasData}
+}
+
+var AreasData = map[string]Area{
+	"Hoofed": {
+		Name: "hoofed", Sectors: map[string]Sector{
+			"Horses": {
+				ID: 1,
+				Animals: []Animal{
+					{ID: 1, Name: "Horse1"},
+					{ID: 2, Name: "Horse2"},
+					{ID: 3, Name: "Horse3"},
+				},
+			},
+			"Cows": {
+				ID: 2,
+				Animals: []Animal{
+					{ID: 4, Name: "Cow1"},
+					{ID: 5, Name: "Cow2"},
+					{ID: 6, Name: "Cow3"},
+				},
+			},
+		},
+	},
+	"Birds": {
+		Name: "birds", Sectors: map[string]Sector{
+			"Doves": {
+				ID: 3,
+				Animals: []Animal{
+					{ID: 7, Name: "Dove1"},
+					{ID: 8, Name: "Dove2"},
+					{ID: 9, Name: "Dove3"},
+				},
+			},
+			"Hawks": {
+				ID: 4,
+				Animals: []Animal{
+					{ID: 10, Name: "Hawk1"},
+					{ID: 11, Name: "Hawk2"},
+					{ID: 12, Name: "Hawk3"},
+				},
+			},
+		},
+	},
+	"Techroom": {
+		Name: "tech room", Sectors: map[string]Sector{
+			"Dining room": {
+				ID: 5,
+				Animals: []Animal{
+					{ID: 13, Name: "Hawk4"},
+				},
+			},
+			"Bath room": {
+				ID: 6,
+				Animals: []Animal{
+					{ID: 14, Name: "Cow4"},
+				},
+			},
+		},
+	},
+}
+
+// Returns the found animal and in which sector and area it is
```

Comment:
Можу підказати, що ось так коменти до експортованих методів пишуться:
```suggestion
// Lookup returns the found animal and in which sector and area it is.
```

Почитати про це ось тут https://go.dev/doc/comment